language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs

```diff
@@ -172,53 +105,26 @@ instance Pretty (Token ann) where
     pretty (LexSpecial _ s)  = pretty s
     pretty EOF{}             = mempty
 
-instance Pretty BuiltinName where
-    pretty AddInteger           = "addInteger"
-    pretty SubtractInteger      = "subtractInteger"
-    pretty MultiplyInteger      = "multiplyInteger"
-    pretty DivideInteger        = "divideInteger"
-    pretty QuotientInteger      = "quotientInteger"
-    pretty ModInteger           = "modInteger"
-    pretty RemainderInteger     = "remainderInteger"
-    pretty LessThanInteger      = "lessThanInteger"
-    pretty LessThanEqInteger    = "lessThanEqualsInteger"
-    pretty GreaterThanInteger   = "greaterThanInteger"
-    pretty GreaterThanEqInteger = "greaterThanEqualsInteger"
-    pretty EqInteger            = "equalsInteger"
-    pretty Concatenate          = "concatenate"
-    pretty TakeByteString       = "takeByteString"
-    pretty DropByteString       = "dropByteString"
-    pretty EqByteString         = "equalsByteString"
-    pretty LtByteString         = "lessThanByteString"
-    pretty GtByteString         = "greaterThanByteString"
-    pretty SHA2                 = "sha2_256"
-    pretty SHA3                 = "sha3_256"
-    pretty VerifySignature      = "verifySignature"
-
-instance Pretty DynamicBuiltinName where
-    pretty (DynamicBuiltinName n) = pretty n
-
-instance Pretty StagedBuiltinName where
-    pretty (StaticStagedBuiltinName  n) = pretty n
-    pretty (DynamicStagedBuiltinName n) = pretty n
-
-instance Pretty TypeBuiltin where
-    pretty TyInteger    = "integer"
-    pretty TyByteString = "bytestring"
-    pretty TyString     = "string"
-
-instance Pretty (Version ann) where
-    pretty (Version _ i j k) = pretty i <> "." <> pretty j <> "." <> pretty k
-
--- | The list of all 'BuiltinName's.
-allBuiltinNames :: [BuiltinName]
-allBuiltinNames = [minBound .. maxBound]
--- The way it's defined ensures that it's enough to add a new built-in to 'BuiltinName' and it'll be
--- automatically handled by tests and other stuff that deals with all built-in names at once.
-
+-- | The list of all 'Keyword's.
 allKeywords :: [Keyword]
 allKeywords = [minBound .. maxBound]
 
--- | The default version of Plutus Core supported by this library.
-defaultVersion :: ann -> Version ann
-defaultVersion ann = Version ann 1 0 0
+-- | An 'IdentifierState' includes a map indexed by 'Int's as well as a map
+-- indexed by 'ByteString's. It is used during parsing.
+type IdentifierState = (M.Map T.Text Unique, Unique)
```

Comment:
`IdentifierState` and its infrastructure was in the `Name` module, but it really only relevant for the lexer, hence it's in the lexer module now. BTW, we could just use `Quote` here and this is what we do in Midnight.

Reply:
:100: that's one of those things I meant to do when I added `Quote` but I think Vanessa didn't like it for some reason.

Reply:
I decided not to update the lexer, because I attempted doing that before and I just don't understand `alex`. It's also the case that `alex` has its own `State` monad (it seems) and Vanessa just put our state into that `State`, which makes sense. I'm not sure if we can stick another monad there.